pkg/odo/genericclioptions/context.go

```diff
@@ -257,3 +260,63 @@ func ignoreButLog(err error) {
 		glog.V(4).Infof("Ignoring error as it usually means flag wasn't set: %v", err)
 	}
 }
+
+// LocalConfigInfo is a struct that will contain the LocalConfig as well as SourcePath
+// the reasoning behind SourcePath is due to the fact that we must correct what the path is
+// on different platforms (Linux, Windows, etc.) as well as get the absolute path of the component.
+type LocalConfigInfo struct {
```

Comment:
This looks a little bit weird to me. Here you are defining another `LocalConfigInfo` with `SourcePath` that overlaps with what is already in `config.LocalConfigInfo` as `SourceLocation`, only in a slightly different format.

Shouldn't we store the SourceLocation  in url format, to make all the conversion easier?

Reply:
I agree.. We could do url format, for example `git:https..blahblah`, `local:` `binary:` etc?

Reply:
I think that a standard url fromat might be better, so for local an binary it would be `file:///path/` or `file:///path/to/file.jar`, and for git `https://github.com/ff/f/` or `git@github.com:user/ff`.

But not sure how `file://` works with a relative path. We will need to do research.

I would like to use something that is a standard rather than creating our own way.

Reply:
it looks like we can't use a schema with the relative file path.


I think that the best would be to just store path for the local and binary component as a relative path with forwarding slashes `/`, Go has functions that deals with OS-specific paths (`/` vs `\`). 
We could use [filepath.FromSlash](https://golang.org/pkg/path/filepath/#FromSlash) and [filepath.ToSlash](https://golang.org/pkg/path/filepath/#ToSlash). ToSlash to convert path to `/` separated, and save it like this. Than before using it call `FromSlash` to convert it to `\`  or `/`  based on the OS.